src/app.ts

```diff
@@ -13,14 +13,37 @@ interface TemperatureSummary {
   average: number
 }
 
+let readingsGlobal: TemperatureReading[]
 export function processReadings(readings: TemperatureReading[]): void {
   // add here your code
-}
 
+  readingsGlobal = readings
+  return
+}
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
   //add here your code
-  return null
+  const filter = readingsGlobal.filter(
+    (data) => data.city === city && data.time.getTime() === date.getTime(),
+  )
+  if (filter.length < 1) {
+    return null
+  }
+  const TemperatureSummary = {
```

Comment:
A common naming convention for functions and variables is camelCase. 
```suggestion
  const temperatureSummary = {
```

Reply:
this was a little mistake I usually do camelCase, fixing it right away